lib/vault/api/kv.rb

```diff
@@ -0,0 +1,207 @@
+require_relative "secret"
+require_relative "../client"
+require_relative "../request"
+require_relative "../response"
+
+module Vault
+  class Client
+    # A proxy to the {KV} methods.
+    # @return [KV]
+    def kv(mount)
+      KV.new(self, mount)
```

Comment:
It's certainly possible, though a little more hassle because we'd have to memoize on a per-mount basis - perhaps that'd be getting carried away with optimisation? I'm certainly happy to make the change though if Hashicorp folks would prefer it.